test/gtest/common/sort.cpp

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+
+using T = int;
+using DV = xhp::distributed_vector<T>;
+using LV = std::vector<T>;
+
+void test_sort(LV &v, auto func) {
+  auto size = v.size();
+  DV d_v(size);
+
+  for (std::size_t idx = 0; idx < size; idx++) {
+    d_v[idx] = v[idx];
+  }
+  barrier();
+
+  std::sort(v.begin(), v.end(), func);
+  xhp::sort(d_v, func);
+
+  barrier();
+  EXPECT_TRUE(equal(v, d_v));
+}
+
+void test_sort2s(LV &v) {
+  test_sort(v, std::less<T>());
+  test_sort(v, std::greater<T>());
```

Comment:
`test_sort` modifies `v`, therefore the second `test_sort` always gets sorted data.
Please take `LV` by value in `test_sort`